acinclude/squid-util.m4

```diff
@@ -214,6 +214,17 @@ AC_DEFUN([SQUID_YESNO],[
   AS_IF([test "$1" != "yes" -a "$1" != "no"],[AC_MSG_ERROR([Bad argument for $2: "$1". Expecting "yes", "no", or no argument.])])
 ])
 
+dnl Check that a library is actually available, useable,
+dnl and where its pieces are (eg headers and hack macros)
+dnl Parameters for this macro are:
+dnl 1) binary library name (without 'lib' prefix)
+dnl 2) logic to run checks
+AC_DEFUN([SQUID_CHECK_LIB_WORKS],[
+AS_IF([m4_translit([test "x$with_$1" != "xno"], [-+.], [___])],[
+  $2
+])
```

Comment:
> PR description: The logic it is given to run should empty the LIBFOO_LIBS variable on detection failures.

Please move this critical "should empty..." expectation from the PR description into the macro documentation inside acinclude/squid-util.m4. I assume that all the converted callers already satisfy this requirement. If not, please do not convert them (or drop the requirement itself for now).

FWIW, it feels like this new macro should standardize how the checks should signal that the library should not be enabled. After that, the macro itself will "empty the LIBFOO_LIBS variable" instead of requiring the caller to do that. However, I assume that you do not want to standardize that at this development stage, and I do not insist on that standardization in this PR. However, now is probably the _best_ time to do it unless you plan on doing the next micro step dedicated to that standardization!

Reply:
FYI, the `LIBFOO_LIBS` variable should be created/set by the logic given to this `SQUID_CHECK_LIB_WORKS` macro. It is the mechanism by which automake links the library. The requirement of setting `LIBFOO_LIBS` empty on failures is to prevent leaking bad values - such as the problem being fixed in PR #1332.

Reply:
I hope [f5221a8](https://github.com/squid-cache/squid/pull/1333/commits/f5221a843a2a1074d8aa1ce62f1d0ea555a1291d) is sufficient.